### PR TITLE
[NEW] Support announcements

### LINKED ISSUE #1593: [NEW] Support "announcement" property
We need to display the *announcement* information in the app.

The announcement needs to appear on the top of the room, showing the property and supporting markdown tags. As an example, you can see how it looks on the web client.

### PR DESCRIPTION
**Pull request by blewjy**
@RocketChat/ios

Closes #1593.

Two main additions to the application:

1. Announcement can be seen on `ChannelActionsView` as one of the channel's information. Placed between Description and Topic, similar to the desktop client.

annoucement-action-view

2. Announcement can also be seen on the main chat view pinned to the top as a banner, similar to the desktop client as well. Implemented as a new `UIView` class, `ChatAnnouncementView`. This banner will follow the colour scheme of the current active theme, with the background being `bannerBackground` and label text `auxiliaryText`. 

Light theme - iPhone XS Max
announcement-light

Dark theme - iPad Pro 12.9 inch (Please ignore the test channels... :P)
announcement-dark-ipad

Black theme - iPhone SE (also a deliberately long announcement to see the text formatting when the label overflows the screen.)
announcement-black-se

Lastly, I have also updated the subscription test to include testing for this new feature.

Please feel free to let me know if there are any changes to be made. Cheers!

### CODE REVIEW

File: Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift [hunk 1]
```diff
@@ -45,6 +46,10 @@ class ChannelActionsViewController: BaseViewController {
                         title: localized("chat.info.item.description"),
                         descriptionText: hasDescription ? subscription.roomDescription : localized("chat.info.item.no_description")
                     ),
+                    ChannelInfoDescriptionCellData(
+                        title: "Announcement",
+                        descriptionText: hasAnnouncement ? subscription.roomAnnouncement : "No announcement"
```
Comment (rafaelks) [inline Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift]: Let's localize the string `"No announcement"` please!

Comment (blewjy) [reply Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift]: Yes, I overlooked this! Am I supposed to add a string to each of the files and Google translate each language? Or do I just add the English version only?

Comment (rafaelks) [reply Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift]: @blewjy Just English is fine, you can just input the English translation in all other languages and you can use **Sem anúncio** for `pt-BR` 💯

File: Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift [hunk 2]
```diff
@@ -45,6 +46,10 @@ class ChannelActionsViewController: BaseViewController {
                         title: localized("chat.info.item.description"),
                         descriptionText: hasDescription ? subscription.roomDescription : localized("chat.info.item.no_description")
                     ),
+                    ChannelInfoDescriptionCellData(
+                        title: localized("chat.info.item.announcement"),
+                        descriptionText: hasAnnouncement ? subscription.roomAnnouncement : localized("chat.info.item.no_announcement")
```
[the remark this answers is not in the document]
Comment (blewjy) [reply Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift]: Corrected.

File: Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift (hunk 1 above)
Comment (blewjy) [reply Rocket.Chat/Controllers/Chat/ChannelActionsViewController.swift]: @rafaelks Done!

**Comment by rafaelks** [comment]
@blewjy Thank you!!! 👍

**Pull request merged**